core/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -144,6 +207,21 @@ private[spark] abstract class YarnSchedulerBackend(
             context.reply(false)
         }
 
+      case c: GetExecutorLossReason =>
+        amEndpoint match {
+          case Some(am) =>
+            Future {
+              context.reply(am.askWithRetry[Option[ExecutorLossReason]](c))
+            } onFailure {
+              case NonFatal(e) =>
+                logError(s"Finding the executor loss reason was unsuccessful", e)
+                context.sendFailure(e)
+            }
+          case None =>
```

Comment:
This is not your fault, but this is starting to get pretty noisy. We need a better way to do this check.